src/document.c

```diff
@@ -1934,7 +1934,12 @@ static gchar *write_data_to_disk(const gchar *locale_filename,
 		if (g_file_set_contents(locale_filename, data, len, &error))
 			geany_debug("Wrote %s with g_file_set_contents().", locale_filename);
 	}
-	else if (USE_GIO_FILE_OPERATIONS)
+	#if defined(_WIN32)
+	#elif defined(_WIN64)
+	#elif defined(__CYGWIN__)
+	#elif defined(__MSYS__)
```

Comment:
I think `_WIN32` is enough to cover all of the above. GLib has one defined though already, it's called `G_OS_WIN32`.